Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Instance/InstanceToTemplateInterface.h

```diff
@@ -46,6 +46,11 @@ namespace AzToolsFramework
             //! @return The string matching the path to the entity alias
             virtual AZStd::string GenerateEntityAliasPath(AZ::EntityId entityId) = 0;
 
+            //! Generates a path to the entity matching the id from the focused prefab.
+            //! @param entityId The entity id to fetch the path for.
+            //! @return The path to the entity with the given id.
+            virtual AZ::Dom::Path GenerateEntityPathFromFocusedPrefab(AZ::EntityId entityId) = 0;
```

Comment:
question: Want to know more about your thoughts of using AZ::Dom::Path here instead of AZStd::string. One point I may already notice is with this Path type we can easily skip an instance from top in the path. Are there other advantages over string?

Reply:
A couple of advantages actually:
1. A string can be any combination of literals whereas with a Dom::Path, we know that the format is going to be a valid path that can be used for looking up in DOMs. It also provides easy-to-use functions like push() and pop() whereas with strings, you would need to find the first '/', do a lot of substring math.
2. AZ::Dom::Path internally uses AZ::Name to store the path tokens and AZ::Name internally uses a dictionary that hashes and ref counts used strings. So it generally results in faster copying and comparing operations.